src/alcazar.rs

```diff
@@ -1,51 +1,88 @@
-use crate::http_request::HttpRequest;
-use std::net::{SocketAddr, TcpListener};
+use crate::{http_request::HttpRequest, router::Router};
+use std::{
+    io::Write,
+    net::{SocketAddr, TcpListener},
+};
 use tracing::info;
 
 #[derive(Default)]
 pub struct AlcazarBuilder {
     url: Option<SocketAddr>,
+    router: Router,
 }
 
 impl AlcazarBuilder {
     pub fn set_addr(self, url: SocketAddr) -> Self {
-        Self { url: url.into() }
+        Self {
+            url: url.into(),
+            router: self.router,
+        }
+    }
+
+    pub fn set_router(self, router: Router) -> Self {
+        Self {
+            url: self.url,
+            router,
+        }
     }
 
     pub fn start(self) -> Alcazar {
         let listener =
             TcpListener::bind(self.url.unwrap_or_else(|| "0.0.0.0:0".parse().unwrap())).unwrap();
 
         let local_addr = listener.local_addr().unwrap();
+        let router = self.router.clone();
 
         info!("listening to {}", local_addr);
         std::thread::spawn(move || loop {
             match listener.accept() {
-                Ok((stream, _addr)) => {
-                    HttpRequest::parse_stream(stream);
+                Ok((mut stream, _addr)) => {
+                    // TODO: Stop to unwrap the world, set up a error handler
+                    let http_request = HttpRequest::parse_stream(&stream).unwrap();
+                    let handler = router.get_handler(
+                        http_request.method().clone().unwrap(),
+                        http_request.path().clone().unwrap(),
+                    );
+                    // TODO: Router and middleware process, early return here for complete response
+                    stream
+                        .write_all(handler.unwrap().clone().get_response().as_bytes())
+                        .unwrap();
+                    stream.flush().unwrap();
                 }
                 Err(_) => info!("Client connexion failed."),
             }
         });
 
-        Alcazar { local_addr }
+        Alcazar {
+            local_addr,
+            router: self.router,
```

Comment:
Does alcazar need a router once it has started ? if not maybe we can remove the clone() call ?

Reply:
The question is, do we need to add some routes during the execution? If not, we don't need it.
Btw if we don't need it actually, there is no reason let it yet. I will remove and wait until a use case.

Reply:
Currently we don't need to have it. All used routes have already prepared for the whole application